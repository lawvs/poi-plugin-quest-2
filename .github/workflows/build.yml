name: Build

on:
  push:
    branches: [main]
  pull_request:
    branches: [main]

jobs:
  build:
    strategy:
      matrix:
        os: [ubuntu-latest]

    if: "!contains(github.event.head_commit.message, '[skip ci]')"
    runs-on: ${{ matrix.os }}

    steps:
      - name: Checkout Repo
        uses: actions/checkout@v3

      - name: Use Node.js
        uses: actions/setup-node@v3
        with:
          node-version-file: '.nvmrc'

      - name: Install node modules
        uses: bahmutov/npm-install@v1

      - name: Build
        run: npm run build

      - name: Lint
        run: npm run lint -- --max-warnings=0

      - name: Type check
        run: npm run typeCheck

      - name: Test
        run: npm run test -- --coverage

      - name: Build storybook
        run: npm run build-storybook -- --quiet

      - name: Deploy GitHub Pages
        if: github.ref == 'refs/heads/main'
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: storybook-static
          force_orphan: true
