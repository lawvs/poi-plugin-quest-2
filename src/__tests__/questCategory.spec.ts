import questCategory from '../../build/questCategory.json'
import { KcwikiQuestData } from '../../build/kcQuestsData'
import { QuestData as KcaQuestData } from '../../build/kcanotifyGamedata'

describe('should questCategory correct', () => {
  test('length', () => {
    expect(questCategory.dailyQuest.length).toMatchInlineSnapshot(`23`)
    expect(questCategory.weeklyQuest.length).toMatchInlineSnapshot(`17`)
    expect(questCategory.monthlyQuest.length).toMatchInlineSnapshot(`11`)
    expect(questCategory.quarterlyQuest.length).toMatchInlineSnapshot(`25`)
    expect(questCategory.yearlyQuest.length).toMatchInlineSnapshot(`44`)
    expect(questCategory.singleQuest.length).toMatchInlineSnapshot(`466`)
  })

  test('snapshot', () => {
    const mergeData = {
      ...KcaQuestData['zh-CN'],
      ...KcwikiQuestData['zh-CN'],
    }
    const humanReadableData = Object.fromEntries(
      Object.entries(questCategory).map(([key, val]) => [
        key,
        val
          .sort((a, b) => a - b)
          .map((gameId) => ({
            gameId,
            code: mergeData[String(gameId) as keyof typeof mergeData].code,
            name: mergeData[String(gameId) as keyof typeof mergeData].name,
          })),
      ])
    )

    expect(humanReadableData).toMatchSnapshot({
      singleQuest: expect.any(Array),
    })
  })
})
