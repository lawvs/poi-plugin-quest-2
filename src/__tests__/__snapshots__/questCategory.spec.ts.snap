// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should questCategory correct snapshot 1`] = `
{
  "dailyQuest": [
    {
      "code": "Bd1",
      "gameId": 201,
      "name": "击破敌舰队！",
    },
    {
      "code": "Bd3",
      "gameId": 210,
      "name": "击破 10 回敌舰队！",
    },
    {
      "code": "Bd4",
      "gameId": 211,
      "name": "击沉三艘敌空母！",
    },
    {
      "code": "Bd6",
      "gameId": 212,
      "name": "打击敌方输送船团！",
    },
    {
      "code": "Bd2",
      "gameId": 216,
      "name": "击破敌舰队主力！",
    },
    {
      "code": "Bd5",
      "gameId": 218,
      "name": "击沉三艘敌方补给舰！",
    },
    {
      "code": "Bd7",
      "gameId": 226,
      "name": "掌握南西诸岛海域制海权！",
    },
    {
      "code": "Bd8",
      "gameId": 230,
      "name": "压制敌方潜水舰！",
    },
    {
      "code": "Cd1",
      "gameId": 303,
      "name": "「演习」锻炼能力！",
    },
    {
      "code": "Cd2",
      "gameId": 304,
      "name": "在「演习」战胜其他提督！",
    },
    {
      "code": "Dd1",
      "gameId": 402,
      "name": "「远征」3 次成功！",
    },
    {
      "code": "Dd2",
      "gameId": 403,
      "name": "「远征」10 次成功！",
    },
    {
      "code": "Ed1",
      "gameId": 503,
      "name": "舰队大整备！",
    },
    {
      "code": "Ed2",
      "gameId": 504,
      "name": "舰队酒保节！",
    },
    {
      "code": "Fd1",
      "gameId": 605,
      "name": "新装备「开发」指令",
    },
    {
      "code": "Fd2",
      "gameId": 606,
      "name": "新造舰「建造」指令",
    },
    {
      "code": "Fd3",
      "gameId": 607,
      "name": "装备「开发」集中強化！",
    },
    {
      "code": "Fd4",
      "gameId": 608,
      "name": "舰娘「建造」舰队强化！",
    },
    {
      "code": "Fd5",
      "gameId": 609,
      "name": "对应裁军条约！",
    },
    {
      "code": "Fd6",
      "gameId": 619,
      "name": "装备的改修强化",
    },
    {
      "code": "Fd7",
      "gameId": 673,
      "name": "装备开发力的整备",
    },
    {
      "code": "Fd8",
      "gameId": 674,
      "name": "工厂环境的整备",
    },
    {
      "code": "Gd1",
      "gameId": 702,
      "name": "实施舰船的「近代化改修」！",
    },
  ],
  "monthlyQuest": [
    {
      "code": "Bm1",
      "gameId": 249,
      "name": "「第五战队」出击！",
    },
    {
      "code": "Bm2",
      "gameId": 256,
      "name": "「潜水舰队」出击！",
    },
    {
      "code": "Bm3",
      "gameId": 257,
      "name": "「水雷战队」前往南西！",
    },
    {
      "code": "Bm4",
      "gameId": 259,
      "name": "「水上打击部队」前往南方！",
    },
    {
      "code": "Bm6",
      "gameId": 264,
      "name": "「空母机动部队」前往西方！",
    },
    {
      "code": "Bm5",
      "gameId": 265,
      "name": "海上护卫强化月",
    },
    {
      "code": "Bm7",
      "gameId": 266,
      "name": "「水上反击部队」突入！",
    },
    {
      "code": "Bm8",
      "gameId": 280,
      "name": "确保后勤线！强化实施海上警备！",
    },
    {
      "code": "Cm1",
      "gameId": 311,
      "name": "精锐舰队演习",
    },
    {
      "code": "Cm2",
      "gameId": 318,
      "name": "给粮舰 “伊良湖” 的支援",
    },
    {
      "code": "Dm1",
      "gameId": 424,
      "name": "强化输送船队护卫！",
    },
    {
      "code": "Fm1",
      "gameId": 626,
      "name": "精锐 “舰战” 队新编成",
    },
    {
      "code": "Fm2",
      "gameId": 628,
      "name": "机种转换",
    },
    {
      "code": "Fm3",
      "gameId": 645,
      "name": "“洋上补给” 物资的调拨",
    },
  ],
  "quarterlyQuest": [
    {
      "code": "Bq11",
      "gameId": 284,
      "name": "南西诸岛方面 “海上警备行动” 发令！",
    },
    {
      "code": "Cq1",
      "gameId": 330,
      "name": "航母机动部队，开始演习！",
    },
    {
      "code": "Cq2",
      "gameId": 337,
      "name": "「十八驱」演习！",
    },
    {
      "code": "Cq3",
      "gameId": 339,
      "name": "「十九驱」演习！",
    },
    {
      "code": "Cq4",
      "gameId": 342,
      "name": "小舰艇群演习强化任务",
    },
    {
      "code": "Dq1",
      "gameId": 426,
      "name": "强化海上通商航路警戒！",
    },
    {
      "code": "Dq2",
      "gameId": 428,
      "name": "压制侵入近海的敌方潜艇！",
    },
    {
      "code": "Fq1",
      "gameId": 637,
      "name": "「熟练搭乘员」养成",
    },
    {
      "code": "Fq2",
      "gameId": 643,
      "name": "主力 “陆攻” 的采购",
    },
    {
      "code": "Fq9",
      "gameId": 653,
      "name": "工厂运转！为下期作战做准备！",
    },
    {
      "code": "Fq3",
      "gameId": 663,
      "name": "继续研究新型舾装",
    },
    {
      "code": "Fq4",
      "gameId": 675,
      "name": "运用装备的综合整备",
    },
    {
      "code": "Fq5",
      "gameId": 678,
      "name": "主力舰上战斗机的更新",
    },
    {
      "code": "Fq6",
      "gameId": 680,
      "name": "对空兵装的整备扩充",
    },
    {
      "code": "Fq7",
      "gameId": 686,
      "name": "战时改修 A 型高角炮的量产",
    },
    {
      "code": "Fq8",
      "gameId": 688,
      "name": "强化航空战力",
    },
    {
      "code": "Bq1",
      "gameId": 822,
      "name": "冲之岛海域迎击战",
    },
    {
      "code": "Bq12",
      "gameId": 845,
      "name": "发令！「西方海域作战」",
    },
    {
      "code": "Bq2",
      "gameId": 854,
      "name": "战果扩张任务！「Z 作战」前段作战",
    },
    {
      "code": "Bq3",
      "gameId": 861,
      "name": "强行运输舰队，拔锚！",
    },
    {
      "code": "Bq4",
      "gameId": 862,
      "name": "实施前线航空侦察！",
    },
    {
      "code": "Bq10",
      "gameId": 872,
      "name": "战果扩张任务！“Z 作战” 后段作战",
    },
    {
      "code": "Bq5",
      "gameId": 873,
      "name": "实施北方海域警备吧！",
    },
    {
      "code": "Bq6",
      "gameId": 875,
      "name": "精锐「三一驱」，突入铁底海域！",
    },
    {
      "code": "Bq7",
      "gameId": 888,
      "name": "新编成「三川舰队」，突入铁底海峡！",
    },
    {
      "code": "Bq8",
      "gameId": 893,
      "name": "彻底确保泊地周边海域的安全！",
    },
    {
      "code": "Bq9",
      "gameId": 894,
      "name": "投入空母战力进行补给线的战斗警戒",
    },
    {
      "code": "Bq13",
      "gameId": 903,
      "name": "扩张「六水战」，向最前线！",
    },
  ],
  "singleQuest": Any<Array>,
  "weeklyQuest": [
    {
      "code": "Bw3",
      "gameId": 213,
      "name": "海上通商破坏作战",
    },
    {
      "code": "Bw1",
      "gameId": 214,
      "name": "あ号作战",
    },
    {
      "code": "Bw2",
      "gameId": 220,
      "name": "い号作战",
    },
    {
      "code": "Bw4",
      "gameId": 221,
      "name": "ろ (Ro) 号作战",
    },
    {
      "code": "Bw5",
      "gameId": 228,
      "name": "海上护卫战",
    },
    {
      "code": "Bw6",
      "gameId": 229,
      "name": "消灭敌东方舰队！",
    },
    {
      "code": "Bw7",
      "gameId": 241,
      "name": "消灭敌北方舰队主力！",
    },
    {
      "code": "Bw8",
      "gameId": 242,
      "name": "击破敌东方中枢舰队！",
    },
    {
      "code": "Bw9",
      "gameId": 243,
      "name": "掌握南方海域珊瑚诸岛的制空权！",
    },
    {
      "code": "Bw10",
      "gameId": 261,
      "name": "努力确保海上输送的安全！",
    },
    {
      "code": "Cw1",
      "gameId": 302,
      "name": "大规模演习",
    },
    {
      "code": "Dw1",
      "gameId": 404,
      "name": "大规模远征作战、宣布！",
    },
    {
      "code": "Dw2",
      "gameId": 410,
      "name": "使向南方的输送作战成功！",
    },
    {
      "code": "Dw3",
      "gameId": 411,
      "name": "继续实施往南方的鼠輸送！",
    },
    {
      "code": "Fw1",
      "gameId": 613,
      "name": "资源的再利用",
    },
    {
      "code": "Fw2",
      "gameId": 638,
      "name": "对空机枪量产",
    },
    {
      "code": "Fw3",
      "gameId": 676,
      "name": "装备开发力的集中整备",
    },
    {
      "code": "Fw4",
      "gameId": 677,
      "name": "续战支援能力的整备",
    },
    {
      "code": "Gw1",
      "gameId": 703,
      "name": "推进「近代化改修」、整顿战备！",
    },
  ],
  "yearlyQuest": [
    {
      "code": "Cy1",
      "gameId": 345,
      "name": "演习的下午茶时间！",
    },
    {
      "code": "Cy2",
      "gameId": 346,
      "name": "精锐至极！主力中的主力，演习开始！",
    },
    {
      "code": "Cy3",
      "gameId": 348,
      "name": "「精锐轻巡」演习！",
    },
    {
      "code": "Cy4",
      "gameId": 350,
      "name": "精锐「第七驱逐队」演习开始！",
    },
    {
      "code": "Cy5",
      "gameId": 353,
      "name": "「巡洋舰战队」演习",
    },
    {
      "code": "Cy6",
      "gameId": 354,
      "name": "「改装特务空母」任务部队演习！",
    },
    {
      "code": "Cy7",
      "gameId": 355,
      "name": "精锐「第十五驱逐队」第一小队演习！",
    },
    {
      "code": "Cy8",
      "gameId": 356,
      "name": "精锐「第十九驱逐队」演习！",
    },
    {
      "code": "Cy9",
      "gameId": 357,
      "name": "「大和型战舰」第一舰队演习开始！",
    },
    {
      "code": "Cy10",
      "gameId": 362,
      "name": "特型初代 “第十一驱逐队” 演习 special！",
    },
    {
      "code": "Cy11",
      "gameId": 368,
      "name": "“十六驱” 演习！",
    },
    {
      "code": "Cy12",
      "gameId": 371,
      "name": "春天到了！「春雨」，进行演习！",
    },
    {
      "code": "Cy13",
      "gameId": 372,
      "name": "实施水上舰 “舰队防空演习”！",
    },
    {
      "code": "Cy14",
      "gameId": 373,
      "name": "【法国舰队】演习！",
    },
    {
      "code": "Dy1",
      "gameId": 434,
      "name": "特设护卫船团司令部，开始活动！",
    },
    {
      "code": "Dy2",
      "gameId": 436,
      "name": "实施练习航海与警备任务！",
    },
    {
      "code": "Dy3",
      "gameId": 437,
      "name": "实行小笠原近海警戒线强化！",
    },
    {
      "code": "Dy4",
      "gameId": 438,
      "name": "实行南西诸岛方面海上护卫强化！",
    },
    {
      "code": "Dy5",
      "gameId": 439,
      "name": "兵站強化远征任务【基本作战】",
    },
    {
      "code": "Dy6",
      "gameId": 440,
      "name": "兵站強化远征任务【扩张作战】",
    },
    {
      "code": "Dy7",
      "gameId": 442,
      "name": "实施西方联络作战准备！",
    },
    {
      "code": "Dy8",
      "gameId": 444,
      "name": "新兵装开发资材输送与船团护卫！",
    },
    {
      "code": "Fy2",
      "gameId": 654,
      "name": "编成精锐双翼机飞行队",
    },
    {
      "code": "Fy3",
      "gameId": 655,
      "name": "工厂完全运作！开发新兵装！",
    },
    {
      "code": "Fy1",
      "gameId": 657,
      "name": "新型兵装的开发整备的强化",
    },
    {
      "code": "Fy4",
      "gameId": 681,
      "name": "航空战力再编成与强化准备",
    },
    {
      "code": "Gy1",
      "gameId": 714,
      "name": "实施「驱逐舰」的改修工事！",
    },
    {
      "code": "Gy2",
      "gameId": 715,
      "name": "续：实施「驱逐舰」的改修工事",
    },
    {
      "code": "Gy3",
      "gameId": 716,
      "name": "实施「轻巡」级改修工事",
    },
    {
      "code": "Gy4",
      "gameId": 717,
      "name": "续： 实施「轻巡」级改修工事",
    },
    {
      "code": "By1",
      "gameId": 904,
      "name": "精锐「十九驱」，闪亮出击！",
    },
    {
      "code": "By2",
      "gameId": 905,
      "name": "「海防舰」，防卫大海！",
    },
    {
      "code": "By3",
      "gameId": 912,
      "name": "工作舰「明石」护卫任务",
    },
    {
      "code": "By4",
      "gameId": 914,
      "name": "重巡战队，向西！",
    },
    {
      "code": "By5",
      "gameId": 928,
      "name": "历战「第十方面舰队」，全力出击！",
    },
    {
      "code": "By6",
      "gameId": 944,
      "name": "实施镇守府近海海域哨戒！",
    },
    {
      "code": "By7",
      "gameId": 945,
      "name": "保护南西方面的运输航路安全！",
    },
    {
      "code": "By8",
      "gameId": 946,
      "name": "空母机动部队，出击！迎击敌舰队！",
    },
    {
      "code": "By9",
      "gameId": 947,
      "name": "AL 作战",
    },
    {
      "code": "By10",
      "gameId": 948,
      "name": "机动部队决战",
    },
    {
      "code": "By11",
      "gameId": 973,
      "name": "日英美联合水上舰队，起航！",
    },
    {
      "code": "By12",
      "gameId": 975,
      "name": "精锐「第十九驱逐队」，全力出击！",
    },
    {
      "code": "By13",
      "gameId": 1005,
      "name": "精强 “第七驱逐队” 紧急出动！",
    },
    {
      "code": "By14",
      "gameId": 1012,
      "name": "鹈来型海防舰，保卫平静的海洋吧！",
    },
    {
      "code": "Fy5",
      "gameId": 1103,
      "name": "量产潜水舰强化兵装",
    },
    {
      "code": "Fy6",
      "gameId": 1104,
      "name": "量产潜水舰电子兵装",
    },
    {
      "code": "Fy7",
      "gameId": 1105,
      "name": "夏日格纳库整备 & 航空基地整备",
    },
    {
      "code": "Fy8",
      "gameId": 1107,
      "name": "【钢材出口】增加陆基航空队的兵力吧！",
    },
    {
      "code": "Fy9",
      "gameId": 1120,
      "name": "【机种整理综合】新型战斗机的量产计划",
    },
    {
      "code": "Fy10",
      "gameId": 1123,
      "name": "增加改良三座水上偵察機",
    },
    {
      "code": "Fy11",
      "gameId": 1138,
      "name": "【高射装置量产】94 式高射装置的追加配备",
    },
  ],
}
`;
